File: mr_utils/recon/reordering/temporal_tv/generate_prior.m
function prior=generate_prior(reduced_k_space)

[sx sy sz]=size(reduced_k_space);

no_lines_first=8;
no_lines_last=8;

% center low res data
center_low_res_data = zeros(size(reduced_k_space));
center_low_res_data(:,1:no_lines_first,:) = reduced_k_space(:,1:no_lines_first,:);
center_low_res_data(:,end-no_lines_last:end,:) = reduced_k_space(:,end-no_lines_last:end,:);

temp_data=zeros(sx,40,sz);
temp_data(:,1:no_lines_first,:)=reduced_k_space(:,1:no_lines_first,:);
temp_data(:,end-no_lines_last:end,:)=reduced_k_space(:,end-no_lines_last:end,:);

filter_hamm=hamming(40);
filtered_temp=zeros(size(temp_data));
for i=1:sz
    for j=1:sx
        filtered_temp(j,:,i)=fftshift(filter_hamm').*squeeze(temp_data(j,:,i));
    end
end

new_center_data=zeros(size(center_low_res_data));
new_center_data(:,1:no_lines_first,:)=filtered_temp(:,1:no_lines_first,:);
new_center_data(:,end-no_lines_last:end,:)=filtered_temp(:,end-no_lines_last:end,:);

low_res_imgs=zeros(size(new_center_data));
for i=1:sz
    low_res_imgs(:,:,i)=fftshift(ifft2(new_center_data(:,:,i)));
end

prior=low_res_imgs;

return;

File: mr_utils/cs/convex/temporal_gd_tv/generate_prior.py
import numpy as np

from mr_utils import view
from mr_utils.load_data import load_mat

def generate_prior(kspace_u):
    '''Hamming filter kspace data to get low-res prior.'''

    sx, _sy, sz = kspace_u.shape[:]

    first = 8
    last = 8+1

    # center low res data
    center_low_res_data = np.zeros(kspace_u.shape, dtype='complex')
    center_low_res_data[:, :first, :] = kspace_u[:, :first, :]
    center_low_res_data[:, -last:, :] = kspace_u[:, -last:, :]

    temp_data = np.zeros((sx, 40, sz), dtype='complex')
    temp_data[:, :first, :] = kspace_u[:, :first, :]
    temp_data[:, -last:, :] = kspace_u[:, -last:, :]

    filter_hamm = np.fft.fftshift(np.hamming(40))
    filtered_temp = temp_data*(
        np.tile(filter_hamm, (sx, sz, 1)).transpose((0, 2, 1)))

    new_center_data = np.zeros(center_low_res_data.shape, dtype='complex')
    new_center_data[:, :first, :] = filtered_temp[:, :first, :]
    new_center_data[:, -last:, :] = filtered_temp[:, -last:, :]

    return np.fft.fftshift(np.fft.ifft2(
        new_center_data, axes=(1, 2)), axes=(1, 2))
